src/directory_listing.rs

```diff
@@ -22,6 +22,21 @@ use std::time::{SystemTime, UNIX_EPOCH};
 
 use crate::{http_ext::MethodExt, Context, Result};
 
+/// Non-alphanumeric characters to be percent-encoded
+/// excluding the "unreserved marks" because allowed in a URI.
+/// See https://www.ietf.org/rfc/rfc2396.txt
+const PERCENT_ENCODE_SET: &AsciiSet = &NON_ALPHANUMERIC
+    .remove(b'_')
+    .remove(b'-')
+    .remove(b'!')
+    .remove(b'.')
+    .remove(b'(')
+    .remove(b')')
+    .remove(b'*')
+    .remove(b'/')
+    .remove(b'\'')
+    .remove(b'~');
```

Comment:
This RFC is obsolete, please refer to https://www.ietf.org/rfc/rfc3986.txt. This means a shorter list of unreserved characters. Also, neither forward nor backward slash belong here (see other comment on how this can be implemented without allowing slashes).